### DISCUSSION TITLE
Starter Assets Character Controllers  How to Use Predictions

### DISCUSSION THREAD

**Discussion by liugoudanzi** [FirstGearGames/FishNet#834] [Feature Request]
![微信截图_20241219131113](https://github.com/user-attachments/assets/1611f960-2073-4a92-b997-a3944e9bdd3b)

fishnet prediction v2 How to integrate into the official project.

I've tried using predictions in a number of ways and failed.

**Comment by DenisFernandes** [comment]
I partially agree with this.

I mean, there's a character controller example already and FPS Land 4.
Seems like the FPS Land development is halted _(and i get it, developing a full authoritative example while maintaining Fish-networking is not easy)_

My take on this is that if the character controller demo that's already included in the package was updated to have a first and third person camera would be a solid starting point for anyone looking to implement networked character controllers

**Comment by FirstGearGames** [comment]
Quite a number of individuals do not like importing all the demos into their projects, and it adds on file size as well. I do agree however such a demo would be very beneficial.

At one point we did have a separate demos git, I think that would be ideal, and simply make that git known within FN.

Also, we could encourage community demos this way. While this is technically not V5 specific, let's put this on the roadmap for V5.

**Reply by Brainshack** [reply]
At least for the asset store version you could previde samples separatly, installable through the package manager. I have seen multiple assets do it this way. This way you dont have to install them at all if you dont want them.
